kkm/kkm_mmu.c

```diff
@@ -349,39 +464,46 @@ int kkm_mmu_sync(uint64_t current_pgd_base, void *guest_kernel_va,
  * walk through kernel page table to identify physical address of faulted address
  * add to to guest kernel and payload page tables
  */
-bool kkm_kontext_mmu_update_priv_area(uint64_t guest_fault_address,
-				      uint64_t monitor_fault_address,
-				      uint64_t current_pgd_base,
-				      struct kkm_mmu_pml4e *guest)
+bool kkm_mmu_update_priv_area(uint64_t guest_fault_address,
+			      uint64_t monitor_fault_address,
+			      uint64_t current_pgd_base,
+			      struct kkm_mmu_pml4e *guest)
 {
 	bool ret_val = true;
 	uint64_t pgd_idx = pgd_index(monitor_fault_address);
+	uint64_t p4d_idx = p4d_index(monitor_fault_address);
 	uint64_t pud_idx = pud_index(monitor_fault_address);
 	uint64_t pmd_idx = pmd_index(monitor_fault_address);
 	uint64_t pte_idx = pte_index(monitor_fault_address);
 	uint64_t table_va = current_pgd_base;
 	uint64_t gva_pte_idx = pte_index(guest_fault_address);
 
-	/* 4th level */
-	if (kkm_kontext_mmu_get_table_va(&table_va, pgd_idx) != true) {
-		printk(KERN_NOTICE
-		       "kkm_kontext_mmu_update_priv_area: pgd failed\n");
+	/* top level */
+	if (kkm_mmu_get_table_va(&table_va, pgd_idx) != true) {
+		printk(KERN_NOTICE "kkm_mmu_update_priv_area: pgd failed\n");
```

Comment:
You could use __FUNCTION__ to have the compiler insert the function name in the string.

Reply:
it is not common in kernel code.